clusterman/kubernetes/kubernetes_cluster_connector.py

```diff
@@ -215,13 +215,13 @@ def _is_node_safe_to_kill(self, node_ip: str) -> bool:
         return True
 
     def _get_nodes_by_ip(self) -> Mapping[str, KubernetesNode]:
-        pool_label_selector = self.pool_config.read_string("pool_label_key", default="clusterman.com/pool")
+        node_label_selector = self.pool_config.read_string("node_label_key", default="clusterman.com/pool")
         pool_nodes = self._core_api.list_node().items
```

Comment:
Wow, I think we can do that as well - https://raw.githubusercontent.com/kubernetes-client/python/master/kubernetes/client/api/core_v1_api.py 

There is some documentation here 

```python3
 :param str label_selector: A selector to restrict the list of returned objects by their labels. Defaults to everything.
    def list_node(self, **kwargs):  # noqa: E501
        """list_node  # noqa: E501

```

Reply:
thoughts and comments @nemacysts @jfongatyelp

Reply:
++ to using a label selector here for the `list_node` call

Reply:
And what is the reason changing label_key setting name?
Deployment will be a bit of a hassle.

Reply:
@ilkinmammadzada current k8s hosts do not have the right label of `paasta.yelp.com/pool`, we added both labels we `yelp.com/pool` and `paasta.yelp.com/pool` to new hosts so we can have backward compatibility . We will remove the old hosts once deployed and then change the key name back.

Reply:
it is fine while migration but you can change the value of the setting instead of setting name. 

this is our setting:
pool_label_key: 'yelp.com/pool'

I mean you can change value of the setting, but not key. hm?

Reply:
We have changed the value of setting here - https://github.yelpcorp.com/sysgit/srv-configs/pull/16438 

and added a new setting to take care of nodes @ilkinmammadzada